lib/rspec/support/fuzzy_matcher.rb

```diff
@@ -6,9 +6,11 @@ module Support
     module FuzzyMatcher
       # @api private
       def self.values_match?(expected, actual)
-        if Hash === actual
-          return hashes_match?(expected, actual) if Hash === expected
-        elsif Array === expected && Enumerable === actual && !(Struct === actual)
+        if actual.instance_of?(Hash)
+          return hashes_match?(expected, actual) if expected.instance_of?(Hash)
+        elsif expected.instance_of?(Array) && actual.instance_of?(Array)
+          return arrays_match?(expected, actual)
+        elsif expected.instance_of?(Array) && Enumerable === actual && !(Struct === actual)
```

Comment:
We avoid `object.instance_of?(klass)` checks, opting for `klass === object` instead.  These should generally be equivalent, but in the case of `object.instance_of?`, since `object` could be anything, we don't know if it will even respond to `instance_of?` or if it's been redefined to do something different.  (Users pass all manner of objects to RSpec...).  For example, `BasicObject === BasicObject.new` returns `true` while `BasicObject.new.instance_of?(BasicObject)` raises a `NoMethodError`.

Please switch all these `instance_of?(klass)` checks to use `klass ===` instead.